lib/mayaUsd/ufe/UsdTRSUndoableCommandBase.h

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright 2020 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include "../base/api.h"
+#include "UsdSceneItem.h"
+
+#include <ufe/transform3dUndoableCommands.h>
+#include <ufe/observer.h>
+
+#include <pxr/usd/usd/attribute.h>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! \brief Base class for translate, rotate, scale undoable commands.
+//
+// UsdRotateUndoableCommand causes USD assertion when used with GfVec3d:
+//
+// Coding Error: in _SetValueImpl at line 5619 of [...]/usd/pxr/usd/usd/stage.cpp -- Type mismatch for </pCylinder1/pCube1.xformOp:rotateXYZ>: expected 'GfVec3f', got 'GfVec3d'
+//
+// Therefore, this class is templated on GfVec3f or GfVec3d, so rotate can use
+// GfVec3f.  To be investigated.  PPT, 9-Apr-2020.
+//
+template<class V>
```

Comment:
Templated on the GfVec type.  Translate is GfVec3d, rotate and scale are GfVec3f, otherwise we crash.  No time to investigate why at this point.